realsense2_camera/src/realsense_node_factory.cpp

```diff
@@ -134,7 +254,12 @@ void RealSenseNodeFactory::onInit()
 #endif
 		ros::NodeHandle nh = getNodeHandle();
 		auto privateNh = getPrivateNodeHandle();
+
+		/* change: sumandeepb: determine correct serial no for realsense_top and realsense_bottom */
+		privateNh.param("camera", _camera_name, std::string(""));
+		privateNh.param("base_frame_id", _tf_link_name, std::string(""));
```

Comment:
should this be base_frame_id or accel_frame_id ?

Reply:
accel_frame_id in TF tree is only available after the sensor is initialized. At this stage of execution only base_frame_id is available. Fortunately this has the same rotation as accel_frame_id just a slightly different translation. Therefore, determining orientation works out.

Reply:
hmm, can you add a todo/Note to the code here to explain the reasoning. Would be helpful to future readers to understand the reason.

Reply:
Added more clear comments to explain the design